src/axolotl/core/trainer_builder.py

```diff
@@ -843,7 +844,14 @@ def build_collator(self, training_args: AxolotlTrainingArguments, **kwargs):
         if self.cfg.model_config_type == "mamba":
             return MambaDataCollator(tokenizer=self.tokenizer)
 
-        return BatchSamplerDataCollatorForSeq2Seq(
+        if training_args.sample_packing:
```

Comment:
I would recommend maybe consolidate the class?

```py
data_collator = BatchSamplerDataCollatorForSeq2Seq if training_args.sample_packing else DataCollatorForSeq2Seq
```

Reply:
<img width="1017" alt="Screenshot 2024-01-08 at 12 11 10 PM" src="https://github.com/OpenAccess-AI-Collective/axolotl/assets/381258/4cefd418-08ec-42d8-b47e-2c2f651ca972">

my IDE doesn't like that 😭